lib/pinch_hitter/message/html.rb

```diff
@@ -0,0 +1,18 @@
+module PinchHitter::Message
+  class Html < ContentType
+
+    def self.valid_message?(string)
+      string =~ /\A(\s|<!doctype[^>]*>)*<html\b/i
```

Comment:
This seems problematic.  Do we need it?

Reply:
Something like this (though not necessarily this particular implementation) is necessary if we need to determine the content type of a response based on its content. It's kind of hard to tell the difference between XML and HTML, since a lot of HTML documents (though not all) would be perfectly valid XML. But if we use another approach, like [you proposed](https://github.com/stevenjackson/pinch_hitter/pull/13#pullrequestreview-357489360), we can avoid the issue and delete these `#valid_message?` methods.